### PR TITLE
Add extra github sauce and tidy up defaults

### PR DESCRIPTION
**Pull request by nabobalis**
## Summary by Sourcery

Update calibration utilities to use GitHub-hosted pointing data, standardize defaults to JSOC, and improve optional handling of correction tables.

Enhancements:
- Switch the master pointing table URL to a GitHub raw file and remove the hardcoded hash
- Standardize default source values to uppercase “JSOC” and allow None to fetch the latest correction data
- Simplify JSOC query construction in get_pointing_table to handle both absent and provided time ranges

Documentation:
- Mark correction_table parameters as optional and update docstrings across calibration and response functions

Tests:
- Add parameterized tests for None and varied casing of the source argument to ensure case-insensitive handling

### CODE REVIEW

**Review by wtbarnes — CHANGES_REQUESTED** [review]
The only real problem I have is `source` being a required keyword argument for `get_pointing_table`.

File: aiapy/calibrate/tests/test_utils.py [hunk 1]
```diff
@@ -43,7 +43,8 @@ def test_correction_table(source) -> None:
         "EFF_AREA",
         "EFF_WVLN",
     ]
-    assert all(cn in table.colnames for cn in expected_columns)
+    for col in expected_columns:
+        assert col in table.colnames
```
[the remark this answers is not in the document]
Comment (nabobalis) [reply aiapy/calibrate/tests/test_utils.py:47]: I just wanted it to error on the specific column instead of all of them.

File: aiapy/calibrate/utils.py [hunk 2]
```diff
@@ -39,8 +39,8 @@
     )
 }
 _URL_HASH_POINTING_TABLE = (
-    "https://aia.lmsal.com/public/master_aia_pointing3h.csv",
-    "a2c80fa0ea3453c62c91f51df045ae04b771d5cbb51c6495ed56de0da2a5482e",
+    "https://raw.githubusercontent.com/LM-SAL/backup_files/refs/heads/aia/generated/aia_pointing_table.csv",
+    None,
```
[the remark this answers is not in the document]
Comment (nabobalis) [reply aiapy/calibrate/utils.py:43]: No, I can use a dummy hash, the "magic" is the context manager.

**Review by wtbarnes — APPROVED** [review]

File: aiapy/calibrate/utils.py [hunk 3]
```diff
@@ -198,11 +198,11 @@ def _get_time(time_range: Time | TimeRange | list | tuple):
     return start, end
 
 
-def get_pointing_table(source="jsoc", time_range=None):
+def get_pointing_table(source="lmsal", time_range=None):
     """
     Retrieve 3-hourly master pointing table from the given source.
 
-    This function can either fetch the pointing table from JSOC or from aia.lmsal.com.
+    This function can either fetch the pointing table from JSOC or from GitHub.
```
[the remark this answers is not in the document]
Comment (nabobalis) [reply aiapy/calibrate/utils.py:205]: True but I guess it is managed by "lmsal" at least in personnel and from the lmsal github org. I can replace with whatever you would like
